src/DataGrid.tsx

```diff
@@ -82,7 +85,7 @@ export interface DataGridProps<R, SR = unknown> extends SharedDivProps {
    * Grid and data Props
    */
   /** An array of objects representing each column on the grid */
-  columns: readonly Column<R, SR>[];
+  columns: readonly (Column<R, SR> | ParentColumn<R, SR>)[];
```

Comment:
Thank you for the POC. 

I can foresee a few issues with this approach. 
1) It cannot handle more that 2 header rows  
2) It relies on unique key to match the parent and the child rows. What happens when a key does not match or column does not have a parent key. 

I think a better api would be use an implicit relationship between header rows so something like 

```
interface ParentColumn {
   name: '',
   children: Array<Column | ParentColumn>;
}

interface Column {
}
```

This way we can specify any level of nesting